modules/launcher/src/main/java/org/mule/module/launcher/DeploymentService.java

```diff
@@ -101,12 +123,29 @@
     void deployDomain(URL domainArchiveUrl) throws IOException;
 
     /**
+     * Deploys a domain bundled as a zip from the given URL to the mule container
+     *
+     * @param domainArchiveUrl location of the zip domain file
+     * @param appProperties the properties to override
+     * @throws IOException
+     */
+    void deployDomain(URL domainArchiveUrl, Properties appProperties) throws IOException;
+
+    /**
      * Undeploys and redeploys a domain
      *
      * @param domainName then name of the domain to redeploy
      */
     void redeployDomain(String domainName);
 
+    /**
+     * Undeploys and redeploys a domain
+     *
+     * @param domainName then name of the domain to redeploy
+     * @param appProperties the properties to override
+     */
+    void redeployDomain(String domainName, Properties appProperties);
```

Comment:
Yes, in that case, it takes the configuration properties from .mule/<app>/configuration-management (it is the file that persists the configuration management properties). In case the redeploy is invoked with new appProperties, the properties in the persisted file will be ignored and overwritten.